### ISSUE TITLE
not working with MQTT based thermostat

### ISSUE THREAD

**Issue by Kingo71** [nielsfaber/zoned-heating#1]
Hi and thank you for the nice project.

I'm using an DIY opentherm thermostat and setup it in HA as below:

`- platform: mqtt
  modes:
   - "off"
   - "heat"
  name: "OpenTherm Thermostat"
  current_temperature_topic: "opentherm-thermostat/current-temperature/get"
  mode_command_topic: "opentherm-thermostat/mode/set"
  mode_state_topic: "opentherm-thermostat/mode/get"
  temperature_command_topic: "opentherm-thermostat/setpoint-temperature/set"
  temperature_state_topic: "opentherm-thermostat/setpoint-temperature/get"
  min_temp: 12
  max_temp: 28
  value_template: "{{ value }}"
  temp_step: 0.5`

If I use it as controller for zoned-heating , this becomes unknown and doesn't work anymore.

Any idea of what can be the issue here?

Thank you in advance.

Angelo.

**Comment by nielsfaber** [comment]
I didn't know anyone was interested in this project, so far I thought I was the only user 😅 
The project is fairly new and I made it for my own use so far, so there could be some compatibility issues in it, but I'd be happy to help fixing them.
I'm using a MQTT climate entity as controller and Tado zone entities, so your controller should be compatible.

It is very strange that the controller gets state `unknown`. 
This usually implies that invalid data is sent over the `mode_state_topic`.
This integration only executes the `set_hvac_mode` and `set_temperature` services on the controller, which cannot cause state to become `unknown`.

Maybe you could create a scenario for reproducing the error?
Does this happen when you turn on a zone?
Is it possible for you to monitor the MQTT data?

You can also enable debug logging for the zoned heating integration, by adding this to the configuration.yaml (and restarting HA):
```
logger:
  default: warning
  logs:
    custom_components.zoned_heating: debug
```
This will provide extra information in the HA logs, so you can follow the steps which are performed, hopefully find out where/if things go wrong.

**Comment by Kingo71** [comment]
Hi Niels, thank you to take some time to reply.

This is a really interesting project indeed! 
In order to manage multiple TVR is a must have.

I did enable the log for your component and this is the only error I get:

ERROR (MainThread) [homeassistant.components.switch] Error adding entities for domain switch with platform zoned_heating

I'm wondering if miss something in the MQTT thermostat.

I did try again and the thermostat state goes to unknown as soon as one of the TVR require the heat.

Thank your in advance for your help and time 

Greetings,
Angelo.
